plugins/modules/accesspoint_workflow_manager.py

```diff
@@ -1624,26 +1671,27 @@ def validate_ap_config_parameters(self, ap_config):
             param_spec = dict(type="str", length_max=32)
             validate_str(ap_config["rf_profile"], param_spec, "rf_profile", errormsg)
 
-        reboot_aps = ap_config.get("reboot_aps")
-        if reboot_aps is not None:
-            ap_identity_type = list(reboot_aps.keys())[0]  # Retrieve the type (mac, hostname, or IP)
-            if ap_identity_type and ap_identity_type in self.keymap and len(reboot_aps.get(ap_identity_type)) > 0:
-                for ap_identifier in reboot_aps[ap_identity_type]:
-                    if ap_identity_type == "mac_addresses":
-                        mac_regex = re.compile(r'^([0-9A-Fa-f]{2}[:-]){5}([0-9A-Fa-f]{2})$')
-                        if not mac_regex.match(ap_identifier):
-                            errormsg.append("mac_addresses: Invalid MAC Address '{0}' in playbook.".format(
-                                str(ap_identifier)))
-                            self.log("Invalid MAC address: {0}".format(ap_identifier), "ERROR")
-                    elif ap_identity_type == "management_ip_addresses" and (not self.is_valid_ipv4(ap_identifier) and
-                                                                            not self.is_valid_ipv6(ap_identifier)):
-                        errormsg.append("management_ip_addresses: Invalid Management IP Address '{0}' in playbook."
-                                        .format(str(ap_identifier)))
-                        self.log("Invalid Management IP address: {0}".format(ap_identifier), "ERROR")
-                    elif ap_identity_type == "hostnames":
-                        param_spec = dict(type="str", length_max=32)
-                        validate_str(ap_identifier, param_spec, "hostnames", errormsg)
-                        self.log("Hostname validation for '{0}' completed.".format(ap_identifier), "DEBUG")
+        for reboot_reset in ("reboot_aps", "factory_reset_aps"):
+            reboot_reset_aps = ap_config.get(reboot_reset)
+            if reboot_reset_aps is not None:
+                ap_identity_type = list(reboot_reset_aps.keys())[0]  # Retrieve the type (mac, hostname, or IP)
+                if ap_identity_type and ap_identity_type in self.keymap and len(reboot_reset_aps.get(ap_identity_type)) > 0:
+                    for ap_identifier in reboot_reset_aps[ap_identity_type]:
+                        if ap_identity_type == "mac_addresses":
+                            mac_regex = re.compile(r'^([0-9A-Fa-f]{2}[:-]){5}([0-9A-Fa-f]{2})$')
+                            if not mac_regex.match(ap_identifier):
+                                errormsg.append("mac_addresses: Invalid MAC Address '{0}' in playbook.".format(
+                                    str(ap_identifier)))
+                                self.log("Invalid MAC address: {0}".format(ap_identifier), "ERROR")
+                        elif ap_identity_type == "management_ip_addresses" and (not self.is_valid_ipv4(ap_identifier) and
```

Comment:
Can we move the invalid IP check inside as like in mac_addresses for better readability?

```
                elif ap_identity_type == "management_ip_addresses":
                    if not self.is_valid_ipv4(ap_identifier) and not self.is_valid_ipv6(ap_identifier):
                        errormsg.append("management_ip_addresses: Invalid Management IP Address '{0}' in playbook."
                                        .format(ap_identifier))
                        self.log("Invalid Management IP address for {0}: {1}".format(reboot_reset, ap_identifier), "ERROR")
                
```